solutions/secure-cross-regional-bucket/main.tf

```diff
@@ -0,0 +1,98 @@
+##############################################################################
+# Secure cross region Bucket
+##############################################################################
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_inputs = var.existing_kms_key_crn == null && var.existing_kms_guid == null ? tobool("A value must be passed for 'existing_kms_guid' if not supplying any value for 'existing_kms_key_crn'.") : true
+
+  bucket_config = [{
+    access_tags                   = var.bucket_access_tags
+    bucket_name                   = var.bucket_name
+    kms_encryption_enabled        = true
+    add_bucket_name_suffix        = var.add_bucket_name_suffix
+    kms_guid                      = var.existing_kms_guid
+    kms_key_crn                   = var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", var.key_ring_name, var.key_name)].crn
+    skip_iam_authorization_policy = var.skip_iam_authorization_policy
+    management_endpoint_type      = var.management_endpoint_type_for_bucket
+    cross_region_location         = var.cross_region_location
+    storage_class                 = var.bucket_storage_class
+    force_delete                  = var.force_delete
+    hard_quota                    = var.hard_quota
+
+    activity_tracking = var.activity_tracker_crn != null ? {
+      read_data_events     = true
+      write_data_events    = true
+      activity_tracker_crn = var.activity_tracker_crn
+    } : null
+    expire_rule = var.expire_days != null ? {
+      enable = true
+      days   = var.expire_days
+    } : null
+    metrics_monitoring = var.monitoring_crn != null ? {
+      usage_metrics_enabled   = true
+      request_metrics_enabled = true
+      metrics_monitoring_crn  = var.monitoring_crn
+    } : null
+    object_versioning = {
+      enable = var.object_versioning_enabled
+    }
+    retention_rule = var.retention_enabled ? {
+      default   = var.retention_default
+      maximum   = var.retention_maximum
+      minimum   = var.retention_minimum
+      permanent = var.retention_permanent
+    } : null
+    cbr_rules = var.bucket_cbr_rules
+  }]
+}
+
+#######################################################################################################################
+# KMS Key
+#######################################################################################################################
+
+# KMS root key for COS cross region bucket
+module "kms" {
+  providers = {
+    ibm = ibm.kms
+  }
+  count                       = var.existing_kms_key_crn != null ? 0 : 1 # no need to create any KMS resources if passing an existing key, or bucket
```

Comment:
DA will be used if bucket needs to create. There is no support for existing bucket. So inline comment can be updated accordingly.